src/webapp/reports/glass-admin/glass-admin-list/atc-classification/useATCActions.tsx

```diff
@@ -0,0 +1,86 @@
+import { useCallback, useState } from "react";
+import { ATCItemIdentifier } from "../../../../../domain/reports/glass-admin/entities/GLASSDataMaintenanceItem";
+import { useAppContext } from "../../../../contexts/app-context";
+import { Namespaces } from "../../../../../data/common/clients/storage/Namespaces";
+import { useSnackbar } from "@eyeseetea/d2-ui-components";
+import i18n from "../../../../../locales";
+
+export function useATCActions(
+    loggerProgram: string,
+    reload: () => void,
+    closePatchModal: () => void,
+    closeUploadATCModal: () => void,
+    closeRecalculateLogicModal: () => void
+) {
+    const { compositionRoot } = useAppContext();
+    const snackbar = useSnackbar();
+
+    const [isRecalculating, setIsRecalculating] = useState<boolean>(false);
+    const [isRecalculated, setIsRecalculated] = useState<boolean>(false);
+
+    const patchVersion = useCallback(
+        async (selectedFile: File | undefined, period: string, selectedItems: ATCItemIdentifier[]) => {
+            try {
+                if (selectedFile) {
+                    await compositionRoot.glassAdmin.uploadFile(Namespaces.ATCS, selectedFile, period, selectedItems);
+                    snackbar.success(i18n.t("Version has been successfully patched"));
+                }
+            } catch (error) {
+                snackbar.error(i18n.t("Error encountered when parsing version"));
+            } finally {
+                closePatchModal();
+                reload();
+            }
+        },
+        [closePatchModal, compositionRoot.glassAdmin, reload, snackbar]
+    );
+
+    const uploadATCFile = useCallback(
+        async (selectedFile: File | undefined, period: string) => {
+            try {
+                if (selectedFile) {
+                    await compositionRoot.glassAdmin.uploadFile(Namespaces.ATCS, selectedFile, period);
+                    snackbar.success(i18n.t("Upload finished"));
+                }
+            } catch (error) {
+                snackbar.error(i18n.t("Error parsing the file"));
+            } finally {
+                closeUploadATCModal();
+                reload();
+            }
+        },
+        [closeUploadATCModal, compositionRoot.glassAdmin, reload, snackbar]
+    );
+
+    const cancelRecalculation = useCallback(async () => {
+        await compositionRoot.glassAdmin
+            .cancelRecalculation(Namespaces.AMC_RECALCULATION)
+            .then(() => setIsRecalculated(false));
+        reload();
+        snackbar.success("Recalculation has been cancelled successfully");
+    }, [compositionRoot.glassAdmin, reload, snackbar]);
+
+    const saveRecalculationLogic = useCallback(async () => {
+        try {
+            setIsRecalculating(true);
+            await compositionRoot.glassAdmin.saveRecalculationLogic(Namespaces.AMC_RECALCULATION, Namespaces.ATCS);
+            snackbar.success(`Please go to the program ${loggerProgram} to see the logs of this recalculation`);
```

Comment:
If here it's the only place where you need `loggerProgram` to show the name in the message, it's better to move  the request from useATCUpload hook and used it like `const loggerProgram = await compositionRoot.glassAdmin.getATCLoggerProgram();`